contracts/PathofTheTemplarShard.sol

```diff
@@ -0,0 +1,130 @@
+pragma solidity ^0.8.13;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "./PartnerMinter.sol";
+
+interface IRelic {
+    function balanceOf(address) external returns (uint256);
+    function tokenOfOwnerByIndex(address, uint256) external returns (uint256);
+    function getRelicInfos(uint256 enclaves) external returns (uint256);
+}
+
+interface IShards {
+    function partnerMint(
+        address account,
+        uint256 id,
+        uint256 amount,
+        bytes memory data
+    ) external;
+}
+
+contract PathofTheTemplarShard is Ownable {
+    address authorizedMinter;
+
+event SignedQuestCompletedMessageHash(bytes32 hash);
+
+    IShards private SHARDS;
+    IRelic private RELIC;
+    uint256[] public SHARD_ID = [1, 2, 3, 4, 5];
+
+    using Counters for Counters.Counter;
+    
+    bytes32 immutable DOMAIN_SEPARATOR;
+
+mapping (msg.sender => bool) public canMint;
+mapping (uint256 => enclaves) signedQuestCompletedMessage;
+
+function mintPathofthetemplarShard() external canMint {
+    SHARDS.partnerMint(msg.sender, SHARD_ID, 1, "");
+}
+
+modifier canMint() {
+    require(msg.sender == authorisedMinter, "You are not allowed to mint");
```

Comment:
where is authorisedMinter? should probably change the canMint mapping on L35 to authorizedMinters, then update this line to `require(authorizedMinters[msg.sender], "Minter forbidden")`

Reply:
Have changed to an error message instead now and ditched the require as consistent with another comment